remoteworkitem/trackeritem_repository.go

```diff
@@ -5,6 +5,14 @@ import "github.com/jinzhu/gorm"
 // upload imports the items into database
 func upload(db *gorm.DB, tqID int, item map[string]string) error {
 	ti := TrackerItem{Item: item["content"], RemoteItemID: item["id"], BatchID: item["batch_id"], TrackerQueryID: uint64(tqID)}
-	err := db.Create(&ti).Error
+	db.Where("remote_item_id = ?", item["id"]).Find(&ti)
+	var err error
+	if ti.ID == 0 {
+		ti1 := TrackerItem{Item: item["content"], RemoteItemID: item["id"], BatchID: item["batch_id"], TrackerQueryID: uint64(tqID)}
+		err = db.Create(&ti1).Error
+	} else {
+		ti2 := TrackerItem{ID: ti.ID, Item: item["content"], RemoteItemID: item["id"], BatchID: item["batch_id"], TrackerQueryID: uint64(tqID)}
```

Comment:
@aslakknutsen What we are dealing here is "tracker items" which can be used to create multiple "work items". Let's keep "tracker items" as a reflection of remote issue tracker.  And move all other complexities into work items.

There is no FK relation between work item and tracker item in the underneath table structure.  The data will be replicated from tracker items to work item multiple times - probably by different users.